sql/wsrep_async_monitor.cc

```diff
@@ -0,0 +1,38 @@
+#ifdef WITH_WSREP
+#include "sql/wsrep_async_monitor.h"
+#include <algorithm>
+#include <cassert>
+
+void Wsrep_async_monitor::enter(seqno_t seqno) {
+  std::unique_lock<std::mutex> lock(m_mutex);
+
+  // Wait for its turn before entering
+  m_cond.wait(lock, [this, seqno]() { return seqno == m_last_left + 1; });
+  m_last_entered = (seqno > m_last_entered) ? seqno : m_last_entered;
```

Comment:
I don't understand this. Is it possible that seqno <= m_last_entered? I think the only possibility enforced by above condition.wait is that seqno grows monolithically.

Reply:
Hi! In an ideal case, it is not possible to have seqno < m_last_entered. I will change it just to `m_last_entered = seqno`